HousingFinanceInterimApi.Tests/V1/UseCase/LoadActionDiaryUseCaseTests.cs

```diff
@@ -0,0 +1,166 @@
+
+public class LoadActionDiaryUseCaseTests
+{
+    private readonly Mock<IBatchLogGateway> _mockBatchLogGateway;
+    private readonly Mock<IBatchLogErrorGateway> _mockBatchLogErrorGateway;
+    private readonly Mock<IActionDiaryGateway> _mockActionDiaryGateway;
+    private readonly Mock<IGoogleFileSettingGateway> _mockGoogleFileSettingGateway;
+    private readonly Mock<IGoogleClientService> _mockGoogleClientService;
+
+    private readonly LoadActionDiaryUseCase _classUnderTest;
+
+    public LoadActionDiaryUseCaseTests()
+    {
+        _mockBatchLogGateway = new Mock<IBatchLogGateway>();
+        _mockBatchLogErrorGateway = new Mock<IBatchLogErrorGateway>();
+        _mockActionDiaryGateway = new Mock<IActionDiaryGateway>();
+        _mockGoogleFileSettingGateway = new Mock<IGoogleFileSettingGateway>();
+        _mockGoogleClientService = new Mock<IGoogleClientService>();
+
+        var waitDurationEnvVar = 30;
+        Environment.SetEnvironmentVariable("WAIT_DURATION", waitDurationEnvVar.ToString());
+
+        _classUnderTest = new LoadActionDiaryUseCase
+            (
+                _mockBatchLogGateway.Object,
+                _mockBatchLogErrorGateway.Object,
+                _mockActionDiaryGateway.Object,
+                _mockGoogleFileSettingGateway.Object,
+                _mockGoogleClientService.Object
+            );
+
+        _mockBatchLogGateway
+            .Setup(g => g.CreateAsync(It.IsAny<string>(), It.IsAny<bool>()))
+            .ReturnsAsync(RandomGen.BatchLogDomain());
+
+        _mockBatchLogGateway
+            .Setup(g => g.CreateAsync(It.IsAny<string>(), It.IsAny<bool>()))
+            .ReturnsAsync(RandomGen.BatchLogDomain());
+
+        var rentPositionFileSettings = RandomGen.CreateMany<GoogleFileSettingDomain>(quantity: 1).ToList();
+        _mockGoogleFileSettingGateway
+            .Setup(g => g.GetSettingsByLabel(It.IsAny<string>()))
+            .ReturnsAsync(rentPositionFileSettings);
+
+        _mockGoogleClientService
+            .Setup(x => x.GetFilesInDriveAsync(It.IsAny<string>(), It.IsAny<string>()))
+            .ReturnsAsync(RandomGen.CreateMany<File>(1).ToList());
+
+        _mockGoogleClientService
+            .Setup(x => x.ReadSheetToEntitiesAsync<ActionDiaryAuxDomain>(
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<string>())
+            ).ReturnsAsync(RandomGen.CreateMany<ActionDiaryAuxDomain>(quantity: 1).ToList());
+    }
+
+    [Fact]
+    public async Task ReturnsStepResponseWhenAllOK()
+    {
+        // Act
+        var useCaseCall = async () => await _classUnderTest.ExecuteAsync().ConfigureAwait(false);
+
+        // Assert
+        await useCaseCall.Should().NotThrowAsync().ConfigureAwait(false);
+
+        var stepResponse = await useCaseCall().ConfigureAwait(false);
+        stepResponse.Continue.Should().BeTrue();
+    }
+
+    [Fact]
+    public async Task LoadsActionDiaryWhenAllOK()
```

Comment:
That's a good happy path test.. from my understanding it pretty much tests if all is expected then the GW calls within the `HandleSpreadSheet` private method all get called.

Though, as you see from the screenshot, this merely tests the try portion of the `HandleSpreadSheet`. So what you'd need to do to have higher tests coverage is test that whenever any of these method throw `ClearActionDiaryAuxiliary`, `CreateBulkAsync`, `LoadActionDiary`, then the `_batchLogErrorGateway.CreateAsync` gets called with correct parameters.

![image](https://github.com/LBHackney-IT/housing-finance-interim-api/assets/43747286/815f97e7-4a08-4000-bc5b-02fa06ccdab3)